protocol/contracts/libraries/Silo/LibSilo.sol

```diff
@@ -8,98 +8,241 @@ pragma experimental ABIEncoderV2;
-        address recipient,
-        uint256 seeds,
-        uint256 stalk
-    ) internal {
-        transferStalk(sender, recipient, stalk);
-        transferSeeds(sender, recipient, seeds);
-    }
-
-    function incrementBalanceOfSeeds(address account, uint256 seeds) internal {
+    /**
+     * @dev Mints Seeds to `account`.
+     */
+    function mintSeeds(address account, uint256 seeds) internal {
         AppStorage storage s = LibAppStorage.diamondStorage();
+
+        // Increase supply of Seeds; Add Seeds to the balance of `account`
         s.s.seeds = s.s.seeds.add(seeds);
         s.a[account].s.seeds = s.a[account].s.seeds.add(seeds);
+
         emit SeedsBalanceChanged(account, int256(seeds));
     }
 
-    function incrementBalanceOfStalk(address account, uint256 stalk) internal {
+    /**
+     * @dev Mints Stalk and Roots to `account`.
+     *
+     * For an explanation of Roots accounting, see {FIXME(doc)}.
+     */
+    function mintStalk(address account, uint256 stalk) internal {
+        AppStorage storage s = LibAppStorage.diamondStorage();
+
+        // Calculate the amount of Roots for the given amount of Stalk.
+        uint256 roots;
+        if (s.s.roots == 0) {
+            roots = uint256(stalk.mul(C.getRootsBase()));
+        } else {
+            roots = s.s.roots.mul(stalk).div(s.s.stalk);
+        }
+
+        // increment user and total stalk
+        s.s.stalk = s.s.stalk.add(stalk);
+        s.a[account].s.stalk = s.a[account].s.stalk.add(stalk);
+
+        // increment user and total roots
+        s.s.roots = s.s.roots.add(roots);
+        s.a[account].roots = s.a[account].roots.add(roots);
+
+
+        emit StalkBalanceChanged(account, int256(stalk), int256(roots));
+    }
+
+
+    /**
+     * @dev mints grownStalk to `account`.
+     * per the zero-withdraw update, if a user plants during the morning,
+     * the roots needed to properly calculate the earned beans would be higher 
+     * than outside the morning. Thus, if a user mows in the morning, 
+     * additional calculation is done and stored for the {plant} function.
+     * @param account farmer
+     * @param stalk the amount of stalk to mint
+     */
+    function mintGrownStalkAndGrownRoots(address account, uint256 stalk) internal {
         AppStorage storage s = LibAppStorage.diamondStorage();
+
         uint256 roots;
-        if (s.s.roots == 0) roots = stalk.mul(C.getRootsBase());
-        else roots = s.s.roots.mul(stalk).div(s.s.stalk);
+        if (s.s.roots == 0) {
+            roots = uint256(stalk.mul(C.getRootsBase()));
+        } else  {
+            roots = s.s.roots.mul(stalk).div(s.s.stalk);
+            if (block.number - s.season.sunriseBlock <= 25) {
+                uint256 rootsWithoutEarned = s.s.roots.add(s.newEarnedRoots).mul(stalk).div(s.s.stalk - (s.newEarnedStalk));
+                uint256 deltaRoots = rootsWithoutEarned - roots;
+                s.newEarnedRoots = s.newEarnedRoots.add(uint128(deltaRoots));
```

Comment:
this `deltaRoots` is the delta for the account stalk amount entered as parameters here. is it intentional that the silo state `newEarnedRoots` is holding a value for an account?